runtime/testy/runtime.py

```diff
@@ -96,16 +99,18 @@ def checkTimedOut(func, *args):
       checkTimedOut(mainFunc)
       nextCall += 1.0/RUNTIME_CONFIG.STUDENT_CODE_HZ.value
       stateQueue.put([SM_COMMANDS.STUDENT_MAIN_OK, []])
-      time.sleep(nextCall - time.time())
-
+      try:
+        time.sleep(nextCall - time.time())
+      except ValueError:
+        nextCall = time.time()
```

Comment:
Nice catch!

Compare with Ansible.py, which does not suffer from this error. (Though it might be a good idea to use the same style here and in Ansible.py, and also grep the code for any more instances of the buggy behavior)